[ts] 6/n tests for PaLM Text Model Parser

Ankush-lastmile opened the pull request:
[ts] 6/n tests for PaLM Text Model Parser




## What

tests for PaLM text model parser

Something to note, not resolved, the palm api for node.js and python seem different.

So what should happen in the config json? Best thing to do here is to just standardize it? pr #400 rfc

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/377).
* __->__ #377
* #396
* #397
* #394
* #376
* #375
* #395

jonathanlastmileai reviewed (commented):
LGTM but stack needs review

saqadri reviewed (approved):
Approving to unblock, some comments

Ankush-lastmile commented on package.json:
```diff
@@ -0,0 +1,6 @@
+{
```
looks like I added types to root of aiconfig, my mistake. Removed

Ankush-lastmile commented on typescript/__tests__/parsers/palm-text/palm-config.json:
updated to suggested name

Ankush-lastmile commented on typescript/__tests__/parsers/palm-text/palm-config.json:
```diff
@@ -0,0 +1,50 @@
+            "metadata": {
+                "model": {
+                    "name": "models/text-bison-001",
+                    "settings": {}
```
Is this referring to the empty objects? If so removed the empty objects

Ankush-lastmile commented on typescript/__tests__/parsers/palm-text/palm.test.ts:
```diff
@@ -0,0 +1,118 @@
+    // no need to instantiate model parser. Load will instantiate it for us since its a default parser
+    const aiConfig = AIConfigRuntime.load(PALM_CONFIG_PATH);
+
+    const completionParams: protos.google.ai.generativelanguage.v1beta2.IGenerateTextRequest = {
```
I couldn't find an exported type for some reason, I expected the same.

Ankush-lastmile commented on typescript/lib/parsers/palm.ts:
```diff
@@ -0,0 +1,205 @@
+
+export class PaLMTextParser extends ParameterizedModelParser {
+  private client: TextServiceClient | null = null;
+  _id = "models/text-bison-001";
```
it sets the id of the Model Parser. Every model parser should have an ID associated with it, for the GPT's we had the parser as a generic and we registered each model with the parser.

PaLM only has 1 model per api so it makes sense to set the id here.

Ankush-lastmile commented:
Addressed all comments. Going to land this stack and can fix forward any other changes that may come up.

Will creating a video demo/testplan to include
